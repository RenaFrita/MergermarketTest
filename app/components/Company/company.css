.company {
  display: flex;
}

.company-name {
  font-size: 20px;
  margin: 0 0 10px;
}

.company-description {
  font-size: 14px;
  margin: 0;
}

.company-details {
  flex: 6;
  padding: 10px;
}

.company-headquarter,
.company-type {
  border-left: 2px solid #000;
  flex: 1;
  font-weight: 700;
  margin: 0;
  padding: 10px;
}
