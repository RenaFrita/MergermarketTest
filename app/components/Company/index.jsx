import React from 'react';
import PropTypes from 'prop-types';
import './company.css';

const Company = ({ name, description, type, headquarter }) => (
  <article className="company">
    <header className="company-details">
      <h2 className="company-name">{name}</h2>
      <p className="company-description">{description}</p>
    </header>
    <p className="company-type">
      {type}
    </p>
    <p className="company-headquarter">
      {headquarter}
    </p>
  </article>
);

Company.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  type: PropTypes.node.isRequired,
  headquarter: PropTypes.string.isRequired
};

export default Company;
