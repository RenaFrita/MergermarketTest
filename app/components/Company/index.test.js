import React from 'react';
import Company from './index.jsx';
import renderer from 'react-test-renderer';

test('Company component markup', () => {
  const component = renderer.create(
    <Company
      name={'Sony'}
      description={'Sony Description'}
      type={'List'}
      headquarter={'USA'}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
