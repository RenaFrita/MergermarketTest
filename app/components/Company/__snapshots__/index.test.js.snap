// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company component markup 1`] = `
<article
  className="company"
>
  <header
    className="company-details"
  >
    <h2
      className="company-name"
    >
      Sony
    </h2>
    <p
      className="company-description"
    >
      Sony Description
    </p>
  </header>
  <p
    className="company-type"
  >
    List
  </p>
  <p
    className="company-headquarter"
  >
    USA
  </p>
</article>
`;
